alcs-frontend/src/app/features/application/applicant-info/application-details/application-details.component.ts

```diff
@@ -21,15 +21,18 @@ export class ApplicationDetailsComponent implements OnInit, OnDestroy {
   @Input() fileNumber!: string;
   @Input() showEdit = false;
   @Input() isSubmitted = true;
+  @Input() wasSubmittedToLfng = false;
 
   authorizationLetters: ApplicationDocumentDto[] = [];
   otherFiles: ApplicationDocumentDto[] = [];
   files: ApplicationDocumentDto[] | undefined;
+  disableEdit = false;
 
   constructor(private applicationDocumentService: ApplicationDocumentService) {}
 
   ngOnInit(): void {
     this.loadDocuments();
+    this.disableEdit = this.wasSubmittedToLfng || !this.isSubmitted;
```

Comment:
Since this is done in ngOninit, if any of these params change this flag won't be recalculated. Move this to ngOnAfterChanges.

Reply:
Fixed!